test/integration/test_integration.py

```diff
@@ -11,54 +14,69 @@
 Result = namedtuple('Result', ['outcome', 'artifact', 'item'])
 
 
-def parse_results(log):
-    '''
-    From the given stdout log, parse the results
-    '''
-    start = 'results:'
-    results = []
-    record = False
+class MockEnv:
+    '''Use this to work with mock. Mutliple instances are not safe.'''
+
+    def __init__(self):
+        self.mock = ['mock', '-r', './mock.cfg']
+        self._run(['--init'], check=True)
+
+    def _run(self, what, **kwargs):
+        return subprocess.run(self.mock + what, **kwargs)
+
+    def copy_in(self, files):
+        self._run(['--copyin'] + files + ['/'], check=True)
+
+    def copy_out(self, directory, *, clean_target=False):
+        if clean_target:
+            with contextlib.suppress(FileNotFoundError):
+                shutil.rmtree(directory)
+        self._run(['--copyout', directory, directory], check=True)
+
+    def shell(self, command):
+        self._run(['--shell', command])
+
+    def orphanskill(self):
+        self._run(['--orphanskill'])
 
-    for line in log.splitlines():
-        if line.strip() == start:
-            record = True
-        if record:
-            results.append(line)
-            if not line:
-                break
 
-    if not results:
-        raise RuntimeError('Could not parse output')
-    return yaml.load('\n'.join(results))['results']
+@pytest.fixture(scope="session")
+def mock():
+    '''Setup a mock we can run Ansible tasks in under root'''
+    mockenv = MockEnv()
+    files = ['taskotron_python_versions'] + glob.glob('*.py') + ['tests.yml']
+    mockenv.copy_in(files)
+    yield mockenv
+    mockenv.orphanskill()
 
 
-def run_task(nevr, *, reterr=False):
+def parse_results(path):
     '''
-    Run the task on a Koji build.
+    From the given result file, parse the results
+    '''
+    with open(path) as f:
+        return yaml.load(f)['results']
+
+
+def run_task(nevr, *, mock):
+    '''
+    Run the task on a Koji build in given mock.
     Returns a dict with Results (outcome, artifact, item)
-    If reterr is true, returns a tuple with the above and captured stderr
-    If reterr is false, prints the stderr
+    Actually returns a tuple with the above and captured log
     '''
-    proc = subprocess.Popen(
-        ['runtask', '-i', nevr, '-t', 'koji_build', 'runtask.yml'],
-        stderr=subprocess.PIPE,
-        universal_newlines=True,
-    )
-    _, err = proc.communicate()
-    if proc.returncode != 0:
-        print(err, file=sys.stderr)  # always print stderr in this case
-        raise RuntimeError('runtask exited with {}'.format(proc.returncode))
-    results = parse_results(err)
+    mock.shell('ansible-playbook tests.yml -e taskotron_item={}'.format(nevr))
+    mock.copy_out('artifacts', clean_target=True)
+
+    with open('artifacts/test.log') as f:
```

Comment:
The log contains all the previous runs, we need to purge it before each test.

Reply:
testing this

Reply:
I might be missing something, but `test.log` should be overwritten in the `Download RPMs from Koji` ansible step. A possible approach is also to use a different artifacts directory for each integration test suite invocation.

Reply:
It contained other stuff. Now I need to preserve each artifacts dir for each nevr, but that's fine. it will also make it easier when debugging test suite failures.

Reply:
The testsuite actually only worked because it got preserved. So this was clearly a bug in the design (previously, each runtask had a different artifact path, now they shared it).

Reply:
Pushed. Also, this may allow us to add a `--fake` switch that will not run the checks but use the stuff from the artifacts directories (e.g. if you are only changing the tests but not the implementation).

Reply:
Added that as well.